From 6aaf4b81953c0f181c0f3c08073ba96ffd6a472e Mon Sep 17 00:00:00 2001
From: Libor Pechacek <lpechacek@suse.com>
Date: Thu, 16 Dec 2021 14:25:56 +0100
Subject: [PATCH] mkdumprd: Properly pass compression params

The compression parameter needs more careful handling so that the
individual bits stay together.

Fixes: 368fbdf62aa8 ("mkdumprd: Do not compress temporary fadump initrd")
Signed-off-by: Libor Pechacek <lpechacek@suse.com>
---
 init/mkdumprd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/init/mkdumprd
+++ b/init/mkdumprd
@@ -86,7 +86,7 @@ function build_initrd()
     if [ "$KDUMP_FADUMP" = "yes" ] ; then
         build_fadumprd
     else
-        run_dracut --compress='xz -0 --check=crc32'
+        run_dracut --compress \'xz -0 --check=crc32\'
     fi
 }                                                                          # }}}
 
